src/upnp.cpp

```diff
@@ -691,13 +691,13 @@ void upnp::try_map_upnp()
 	TORRENT_ASSERT(is_single_thread());
 	if (m_devices.empty()) return;
 
-	for (auto i = m_devices.begin(), end(m_devices.end()); i != end; ++i)
+	for (const auto& m_device : m_devices)
```

Comment:
sorry for a nit-pick, but I use "east const" int this code base. i.e. `auto const&`